libs/gitignore/Gitignore_filter.ml

```diff
@@ -79,6 +84,8 @@ let select_path opt_gitignore_file_cache sel_events levels relative_segments =
           | []
           | [ "" ] ->
               loop sel_events levels file_path segments
+          (* If a path has been deselected, don't test for dir-only patterns *)
+          | _ :: _ when deselected -> loop sel_events levels file_path segments
```

Comment:
would be good to have a unit test for that. @mjambon wrote many of them in Unit_gitignore.ml

Reply:
Will add one

Reply:
Also, if you see any spots where tests are missing, don't hesitate to add a `(* TODO: test this *)`. It's useful for whomever will work on the relevant test suite later and acknowledges that the situation isn't ideal.